Formula/python@2.rb

```diff
@@ -175,18 +156,7 @@ def install
     args << "CPPFLAGS=#{cppflags.join(" ")}" unless cppflags.empty?
 
     system "./configure", *args
-
-    # HAVE_POLL is "broken" on macOS. See:
-    # https://trac.macports.org/ticket/18376
-    # https://bugs.python.org/issue5154
-    if build.without? "poll"
-      inreplace "pyconfig.h", /.*?(HAVE_POLL[_A-Z]*).*/, '#undef \1'
-    end
```

Comment:
If the poll option is vanishing, then this will be built as if `--without-poll` were passed, which means we still need the inreplace, right?

Reply:
After reviewing the history of this hack, we want to get rid of it (i.e. in effect make `--with-poll` the new behavior).

Some background: the `poll()` system call on macOS is limited in some cases (devices, rather than regular files). Someone reported it to python, and suggested turning it off completely, i.e. mimicking a system that has no poll support at all. Upstream considered it, and decided they’d rather ship with poll support “as close to the OS”, rather than disable it altogether. We should follow that decision.